unittests/launch_box/test_launch_box.cu

```diff
@@ -0,0 +1,36 @@
+#include <iostream>
+
+#include <gunrock/cuda/launch_box.hxx>
+
+using namespace gunrock::cuda;
+
+int main(void) {
+
+  // First template argument signifies the SM version
+  typedef launch_box_t<
+    sm_launch_params_t<86, 16, 64, 2>,
+    sm_launch_params_t<80, 16, 32, 4>,
+    sm_launch_params_t<75, 32, 64, 8>,
+    sm_launch_params_t<35, 64, 64, 16>
+  > launch_t;
+
+  // They also have a short type name
+  typedef launch_box_t<
+    sm<86, 16, 64, 2>,
+    sm<80, 16, 32, 4>,
+    sm<75, 32, 64, 8>,
+    sm<35, 64, 64, 16>
```

Comment:
Let's do `sm_t` for this.

Reply:
nervous about specifying the same data in multiple different places. what if they are out of sync?

Reply:
These are all just 3 different aliases for launch params that I wanted to show. Each if the 3 launch boxes is identical at runtime. They aren't dependent on each other in any way.

Reply:
What I'm saying is that you have to type the same data three different times, which is not what you'd prefer to do; you'd prefer to do it once, because it opens the possibility of "someone makes a change to one of the identical items but not the other two" and that would be a really nasty bug to track down. If I'm not reading this right, tell me.

Reply:
I understand what you're saying. I think this example is confusing.

You only need to specify you're data in one place. A user would pick any one of these three methods when defining their launch box.

For the sake of the unit test I was just trying to show off all three ways of using it. When it gets actually implemented we would just pick one of these three (I personally like the 2nd one best).

Reply:
Might be nice to actually document an example elsewhere in comments. This is a unit test and it is supposed to do all of this (even if we are only using one of them).

Reply:
Understood. I can write up some test cases when I am finishing it off.